test/e2e/persistent_volumes.go

```diff
@@ -66,58 +66,96 @@ var _ = framework.KubeDescribe("PersistentVolumes", func() {
-		framework.WaitForPersistentVolumePhase(api.VolumeBound, c, pv.Name, 1*time.Second, 30*time.Second)
+		if err != nil {
+			framework.Failf("PersistentVolumeClaim errored during creation: %v", err)
+		}
 
+		// Wait for PersistentVolume.Status.Phase to be Bound. Can take several minutes.
+		err = framework.WaitForPersistentVolumePhase(api.VolumeBound, c, pv.Name, 3*time.Second, 300*time.Second)
+		if err != nil {
+			framework.Failf("PersistentVolume did not enter a bound state: %+v", err)
+		}
+		// Check the PersistentVolume.ClaimRef.UID for non-nil value as confirmation of the bound state.
+		framework.Logf("Checking ClaimRef value of the PersistentVolume")
 		pv, err = c.PersistentVolumes().Get(pv.Name)
-		Expect(err).NotTo(HaveOccurred())
-		if pv.Spec.ClaimRef == nil {
-			framework.Failf("Expected PersistentVolume to be bound, but got nil ClaimRef: %+v", pv)
+		if pv.Spec.ClaimRef == nil || len(pv.Spec.ClaimRef.UID) == 0 {
+			pvJson, _ := json.MarshalIndent(pv, "", "  ")
+			framework.Failf("Expected PersistentVolume to be bound, but got nil ClaimRef: %+v", string(pvJson))
+		}
+
+		// Check the PersistentVolumeClaim.Status.Phase for Bound state
+		pvc, err = c.PersistentVolumeClaims(ns).Get(pvc.Name)
+		framework.Logf("Checking the bind status of the PersistentVolumeClaim: %+v", pvc.Name)
+		if pvcPhase := pvc.Status.Phase; pvcPhase != "Bound" {
+			framework.Failf("Expected PersistentVolumeClaim phase Bound.  Actual phase:  %+v.  Error: %+v", pvcPhase, err)
+		}
+
+		// Checkpod runs a command to both write to the volume and read a prewritten index.html file on the volume.
+		// If both actions succeed, exits 0.  Else exits 1
+		pvc, _ = c.PersistentVolumeClaims(ns).Get(pvc.Name)
+		checkpod := makeCheckPod(ns, pvc.Name)
+		checkpod, err = c.Pods(ns).Create(checkpod)
+		if err != nil {
+			framework.Failf("Error during checkpod create: %+v", err)
 		}
 
+		// Wait for the checkpod to complete it's lifecycle
+		err = framework.WaitForPodSuccessInNamespace(c, checkpod.Name, checkpod.Spec.Containers[0].Name, checkpod.Namespace)
+		if err != nil {
+			framework.Failf("Pod failed during creation: %+v", err)
+		}
+
+		// Delete the PersistentVolumeClaim
 		framework.Logf("Deleting PersistentVolumeClaim to trigger PV Recycling")
 		err = c.PersistentVolumeClaims(ns).Delete(pvc.Name)
-		Expect(err).NotTo(HaveOccurred())
+		if err != nil {
+			framework.Failf("PersistentVolumeClaim delete experienced an unexpected error: %v", err)
+		}
 
-		// allow the recycler a chance to catch up.  it has to perform NFS scrub, which can be slow in e2e.
-		framework.WaitForPersistentVolumePhase(api.VolumeAvailable, c, pv.Name, 5*time.Second, 300*time.Second)
+		// Wait for the PersistentVolume phase to return to Available
+		framework.Logf("Waiting for recycling process to complete.")
+		err = framework.WaitForPersistentVolumePhase(api.VolumeAvailable, c, pv.Name, 3*time.Second, 300*time.Second)
```

Comment:
Ya, my problem is probably more related to the fact that the pv's claimRef isn't nil'd out so maybe that causes the Phase to change to _Released_?  Since the original test tested for _Available_ and your test works I'd leave it as is.